rapids/rapids.sh

```diff
@@ -123,15 +92,55 @@ function execute_with_retries() {
   return 1
 }
 
+readonly conda_env="/opt/conda/miniconda3/envs/dask-rapids"
 function install_dask_rapids() {
-  if is_debian11 || is_debian12 || is_ubuntu20 || is_ubuntu22 ; then
-      local python_ver="3.10"
-  else
-      local python_ver="3.9"
+  if is_cuda12 ; then
+    local python_ver="3.10"
+    local cuda_spec="cuda-version>=12,<=12.5"
+    local dask_spec="dask"
+    local numba_spec="numba"
+  elif is_cuda11 ; then
+    local python_ver="3.9"
+    local cuda_spec="cuda-version>=11,<11.6"
```

Comment:
I recall an error that caused me to put this limit in here.  It may be too strict.  I'll try relaxing it once the tests are passing again.

Reply:
> Would it be possible to allow CUDA 11.8 here?
> 
> ```suggestion
>     local cuda_spec="cuda-version>=11,<12"
> ```

I did <=11.8 instead of <12 for now.  Once the tests are green, I'll try being more lax

Reply:
Continuing in this thread: https://github.com/GoogleCloudDataproc/initialization-actions/pull/1219#discussion_r1720297164